src/west/app/project.py

```diff
@@ -640,6 +644,46 @@ def _die_if_manifest_project_filter(self, action):
                      'the manifest while projects are made inactive by the '
                      'project filter.')
 
+    def _status(self):
+        ppaths = []
+        untracked = []
+        for project in self._projects(None):
+            # Skip inactive projects unless the user said
+            # --all or named some projects explicitly.
+            if not self.manifest.is_active(project):
+                self.dbg(f'{project.name}: skipping inactive project')
+                continue
+            ppaths.append(Path(project.abspath).resolve())
+
+        def _folder_status(folder):
+            self.dbg(f'folder status in: {folder}')
+            for e in [e.resolve() for e in folder.iterdir()]:
+                if e.is_dir():
+                    self.dbg(f'processing folder: {e}')
+                    for ppath in ppaths:
+                        if e.samefile(ppath):
+                            # We hit a project root folder, skip it
+                            break
+                        elif e in ppath.parents:
+                            self.dbg(f'recursing into: {e}')
+                            _folder_status(e)
+                            break
+                    else:
+                        # No match, untracked folder
+                        untracked.append(e)
+                        continue
+                else:
+                    # Untracked file
+                    untracked.append(e)
+
+        # Avoid using Path.walk() since that returns all files and folders under
+        # a particular folder, which is overkill in our case. Instead, recurse
+        # only when required.
+        _folder_status(Path.cwd())
```

Comment:
Why not `topdir`? I think it would be confusing to have different output depending on where you are in the workspace, also because I would hope the paths are relative to `topdir` in the output?

Reply:
Because this is how `git status` behaves. This is another thing I wanted to discuss

Reply:
The discussion was whether to find untracked files from `cwd()` and below or from `topdir` and below (i.e. the whole workspace) we ended up with the latter because that's how `git status` works for untracked files.